balancer/balancer.go

```diff
@@ -187,9 +184,19 @@ func (b *BalancerImpl) execBalancer(balancerArgs []string) error {
 		return err
 	}
 
-	err = b.cmd.Wait()
+	go func() {
+		defer cancel()
+		err = b.cmd.Wait()
+	}()
 
-	return err
+	<-ctx.Done()
+
+	if err != nil {
+		return err
+	}
+
+	glog.Infof("killing MistUtilLoad")
+	return b.cmd.Process.Kill()
```

Comment:
It was needed before and we frequently leaked MistUtilLoad processes, whoops. But until this refactor it was difficult to get a proper exit handler going because we were fighting with the Serf CLI. FIxed now.